backend/app.ts

```diff
@@ -69,6 +69,16 @@ export function createFrontend(
       color: instance.color,
     });
   });
+  app.delete("/instances/:id", (req, res) => {
+    instances.delete(parseInt(req.params.id));
```

Comment:
I think it's time to extract what's in common with .get() here into a separate method and use it in both places.